header.c

```diff
@@ -831,7 +836,11 @@ int sam_hdr_update_target_arrays(sam_hdr_t *bh, const sam_hrecs_t *hrecs,
             if (!bh->target_name[i])
                 return -1;
         }
-        bh->target_len[i] = hrecs->ref[i].len;
+        if (hrecs->ref[i].len < UINT32_MAX) {
+            bh->target_len[i] = hrecs->ref[i].len;
+        } else {
+            bh->target_len[i] = UINT32_MAX;
```

Comment:
It shouldn't need to because `sam_hdr_tid2len()` uses the `sam_hrecs_t::ref` array if available.  So if `target_len` is being updated from `sam_hrecs_t::ref`, `sdict` isn't going to be used any more and arguably it could actually be freed at this point.